builtin/aws/ecs/platform.go

```diff
@@ -2389,14 +2442,16 @@ type Config struct {
 	// The environment variables to pass to the main container
 	Environment map[string]string `hcl:"static_environment,optional"`
 
-	// The secrets to pass to to the main container
+	// The secrets to pass to the main container
 	Secrets map[string]string `hcl:"secrets,optional"`
 
-	// Assign each task a public IP. Default false.
-	// TODO to access ECR you need a nat gateway or a public address and so if you
-	// set this to false in the default subnets, ECS can't pull the image. Leaving
-	// it disabled until we figure out how to handle that onramp case.
-	// AssignPublicIp bool `hcl:"assign_public_ip,optional"`
+	// Assign each task a public IP. Default true.
+	// Note: If private subnets have been specified for ECS tasks, and
+	// no NAT gateway is configured, ECS will be unable to pull your image
+	// from ECR and your services will be unable to start.
+	// NOTE(izaak): If this is not set, we can probably set it to true if
+	// the ecs subnet's route table has a nat gw default route, and true if they don't.
+	AssignPublicIp *bool `hcl:"assign_public_ip,optional"`
```

Comment:
Using a *bool so we can potentially set a clever default in the future if this value is unset.